### DISCUSSION TITLE
how to display the node index number on the field data?

### DISCUSSION THREAD

**Discussion by zhifeiji-han** [SCIInstitute/SCIRun#2306] [General]
I draw a graph to show the geometric data of the torso field in SCIRun.
 I want to see the order of the index numbers of the geometric nodes.
Is there any module that can realize this function?

**Comment by zhifeiji-han** [comment]
![image](https://user-images.githubusercontent.com/70640243/124474520-33b0e100-ddd3-11eb-8505-0a71f6b15cd8.png)

**Comment by tpat95** [comment]
First, make the value of the nodes the index through CreateFieldData. You'll probably want to scale that otherwise the large indices will be too big for the scene. Then set the nodes in ShowField to spheres so they can be vary in size. You should now see the nodes varying in size in the order of indices.

When you do that first part, you can apply a color map to the nodes to make it easier to see.

In the future, you'll be able to do this with text rendering. 

Also, please put usage questions on the discussions page.

**Comment by zhifeiji-han** [comment]
Thank you for your reply！
But I want the index number of the node, not the value of the node.
I can do it on Map3D, but I don't know how to do it on scirun.
![image](https://user-images.githubusercontent.com/70640243/124529299-d3ef2000-de3c-11eb-9ac2-d1670d44a9e9.png)

**Comment by tpat95** [comment]
Visualizing indices directly like that won't be possible until text rendering is implemented.

What meant is that you can set the index to the node value. Here is an image where a color map is applied to a field where the index is set as the value. 
![image](https://user-images.githubusercontent.com/25259698/124662525-314b9700-de66-11eb-8f88-3e7f1aee6510.png)

It doesn't have the accuracy that text rendering does, but hopefully this is helpful until we have support for this.

Here is the network I used to make the image. You'll have replace the path of the object file to something on your computer.
[index_viz.zip](https://github.com/SCIInstitute/SCIRun/files/6772679/index_viz.zip)

**Comment by zhifeiji-han** [comment]
Thank you for your reply. I see what you mean.

**Comment by dcwhite** [comment]
Here's the issue: https://github.com/SCIInstitute/SCIRun/issues/2148
Not sure when we'll get to that one.
